### DISCUSSION TITLE
Locale specific site preferences for multi locale site??

### DISCUSSION THREAD

**Discussion by gania241** [SalesforceCommerceCloud/sfcc-ci#410] [Team Posts]
Hi Folks,
What is the best way to store/retrieve locale specific site preferences for multi locale site without using any custom object JSON?

**Comment by pravngaur** [comment]
Hello Prasanna,

As you know, locale specific site preferences are not supported in
platform, i am afraid you would need to use custom objects.

You can write custom logic to store specific formatted json in preferences
and then parsing it meaningfully but then it would be reinventing the wheel
and moreover it would be costly to maintain custom solution.

So custom object i would recommend.

**Comment by gania241** [comment]
@pravngaur, Keeping mind the quote limit ,the site data (more than 5 locales in a single site) and usability my client has no interest to store a data in a huge JSON so looking for an alternative..
NOTE : Site pref JSON won't work for types such as image, date, boolean, enum of strings..

**Comment by pravngaur** [comment]
Could you please explain the use case?

If this is not one time configuration but the data is ever growing then you
are correct, you must be mindful of quota and governance limits.

I am afraid there is no alternative per se but you need to have the
automated house keeping jobs while using custom objects.

**Comment by zlatinz** [comment]
How many of these settings the customer will need to change more than once an year?

What are these settings about, can the solution be done using some alternative approach 
1. Think of properties files in the webdav folder for Velocity templates as one option, should it indeed be something that the customer may need to customise, and yet it is huge number of settings with locale specific ones
2. Using JSON files (one per locale) in the cartridge can be another option, as you can require a json file from script and you would get access to all the data within the JSON string. This would not allow for dynamic editing but it would be more structured way for achieving the task.
3. Also consider if any of these are related to integration, and if you can use Service Framework settings for these (including custom attributes e.g. on service credentials) and Job Framework Parameters.

Having large list of site preferences in my experience has proven to be a bottleneck, and at some point the fear of breaking stuff raises exponentially and no settings are changed or deleted, even though they may no longer be applicable.
